### PR TITLE
Add gomod test for CAPM3 and IPAM main/stable branches

### PR DESCRIPTION
**Pull request by kashifest**

### CODE REVIEW

File: prow/config/config.yaml [hunk 1]
```diff
@@ -365,6 +365,22 @@ presubmits:
         imagePullPolicy: Always
 
   metal3-io/cluster-api-provider-metal3:
+  - name: gomod
+    branches:
+    - main
```
Comment (furkatgofurov7) [inline prow/config/config.yaml:370]: can you please add "release-1.1" to this list?

File: prow/config/config.yaml [hunk 2]
```diff
@@ -651,6 +667,22 @@ presubmits:
         imagePullPolicy: Always
 
   metal3-io/ip-address-manager:
+  - name: gomod
+    branches:
+    - main
```
Comment (furkatgofurov7) [inline prow/config/config.yaml:672]: same comment as above

**Review by furkatgofurov7 — CHANGES_REQUESTED** [review]

File: prow/config/config.yaml (hunk 1 above)
Comment (kashifest) [reply prow/config/config.yaml:370]: I haven’t added the gomod file in that branch yet.

Comment (furkatgofurov7) [reply prow/config/config.yaml:370]: yes, but we can add that branch with this PR and backport the https://github.com/metal3-io/cluster-api-provider-metal3/pull/725 to release-1.1 branch once merged

File: prow/config/config.yaml (hunk 2 above)
Comment (kashifest) [reply prow/config/config.yaml:673]: Done

File: prow/config/config.yaml (hunk 1 above)
Comment (kashifest) [reply prow/config/config.yaml:370]: Done

**Review by furkatgofurov7 — APPROVED** [review]

**Comment by kashifest** [comment]
cc @lentzi90 @smoshiur1237 @mboukhalfa

**Review by smoshiur1237 — COMMENTED** [review]
/lgtm

**Pull request merged**